modules/mod-deep-learning/ModelCard.cpp

```diff
@@ -0,0 +1,371 @@
+   writer.String(m_name.c_str());
+
+   // author
+   writer.String("author");
+   writer.String(m_author.c_str());
+   
+   // long description
+   writer.String("long_description");
+   writer.String(m_long_description.c_str());
+
+   // short description
+   writer.String("short_description");
+   writer.String(m_short_description.c_str());
+
+   // sample rate
+   writer.String("sample_rate");
+   writer.Int(m_sample_rate);
+
+   // multichannel
+   writer.String("multichannel");
+   writer.Bool(m_multichannel);
+
+   // effect type
+   writer.String("effect_type");
+   writer.String(m_effect_type.c_str());
+
+   // domain tags
+   writer.String("domain_tags");
+   writer.StartArray();
+   for (auto tag : m_domain_tags)
+      writer.String(tag.c_str());
+   writer.EndArray();
+
+   // tags
+   writer.String("tags");
+   writer.StartArray();
+   for (auto tag : m_tags)
+      writer.String(tag.c_str());
+   writer.EndArray();
+
+   // labels
+   writer.String("labels");
+   writer.StartArray();
+   for(auto label : m_labels)
+      writer.String(label.c_str());
+   writer.EndArray();
+
+   // model size
+   writer.String("model_size");
+   writer.Uint64((uint64_t)m_model_size);
+
+}
+
+void ModelCard::Deserialize(const Doc& doc, const Doc& schema)
+{
+   using namespace validators;
+   try
+   {
+      Validate(doc, schema);
+   }
+   catch(const InvalidModelCardDocument& e)
+   {
+      wxLogError(wxString(e.what()));
+      return;
+   }
+
+   // these fields are not in HF mettadata but rather added later,
+   // so don't throw if they are not present
+   m_author = tryGetString("author", doc, "");
+   m_name = tryGetString("name", doc, "");
+   m_model_size = (size_t)tryGetUint64("model_size", doc, 0);
+
+   m_long_description = tryGetString("long_description", doc, 
+                                       "no long description available");
+   m_short_description = tryGetString("short_description", doc, 
+                                       "no short description available");
+   m_effect_type = toLowerCase(tryGetString("effect_type", doc));
+   m_domain_tags = tryGetStringArray("domain_tags", doc, true);
+   m_tags = tryGetStringArray("tags", doc);
+   m_labels = tryGetStringArray("labels", doc);
+   m_sample_rate = tryGetInt("sample_rate", doc);
+   m_multichannel = tryGetBool("multichannel", doc);
+}
+
+std::string ModelCard::GetRepoID() const
+{
+   return this->author() + '/' + this->name();
+}
+
+bool ModelCard::operator==(const ModelCard& that) const
+{
+    return GetRepoID() == that.GetRepoID();
+} 
+
+bool ModelCard::operator!=(const ModelCard& that) const
+{
+    return !((*this) == that);
+}
+
+// ModelCardCollection implementation
+
+
+void ModelCardCollection::Insert(ModelCardHolder card)
+{  
+    // only add it if its not already there
+   auto it = std::find_if(this->begin(), this->end(), [&](ModelCardHolder a){
+      return (*card) == (*a);
+   });
+
+   if (it == this->end())
+      mCards.push_back(card);
```

Comment:
Might `move(card)` for another tiny efficiency, or make the argument of `Insert()` a const reference, avoiding one shared pointer copy